README.md
# ØKP4 Faucet Web App

> 🚰 Token Faucet Web App for the [ØKP4]([okp4.network](https://okp4.network)) testnet network.

[![version](https://img.shields.io/github/v/release/okp4/faucet-web?style=for-the-badge&logo=github)](https://github.com/okp4/faucet-web/releases)
[![lint](https://img.shields.io/github/actions/workflow/status/okp4/faucet-web/lint.yml?branch=main&label=lint&style=for-the-badge&logo=github)](https://github.com/okp4/faucet-web/actions/workflows/lint.yml)
[![build](https://img.shields.io/github/actions/workflow/status/okp4/faucet-web/build.yml?branch=main&label=build&style=for-the-badge&logo=github)](https://github.com/okp4/faucet-web/actions/workflows/build.yml)
[![conventional commits](https://img.shields.io/badge/Conventional%20Commits-1.0.0-yellow.svg?style=for-the-badge&logo=conventionalcommits)](https://conventionalcommits.org)
[![contributor covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg?style=for-the-badge)](https://github.com/okp4/.github/blob/main/CODE_OF_CONDUCT.md)
[![license](https://img.shields.io/badge/License-BSD_3--Clause-blue.svg?style=for-the-badge)](https://opensource.org/licenses/BSD-3-Clause)
![TypeScript](https://img.shields.io/badge/typescript-%23007ACC.svg?style=for-the-badge&logo=typescript&logoColor=white)
[![prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=for-the-badge)](https://github.com/prettier/prettier)

## Purpose

A webapp that allows users to get a small amount of `knøw` tokens to play with the [ØKP4]([okp4.network](https://okp4.network)) testnet.

## Setup

🚚 Install the dependencies and build the project:

```sh
yarn

yarn build
```

### Launch

Run the server with the following command line:

```sh
yarn start
```

Then open [http://localhost:3000](http://localhost:3000) to launch the Web App.


CHANGELOG.md
# ØKP4 Faucet-Web changelog

## [1.2.1](https://github.com/okp4/faucet-web/compare/v1.2.0...v1.2.1) (2023-06-14)


### Bug Fixes

* bump okp4 ui version ([1bb74f1](https://github.com/okp4/faucet-web/commit/1bb74f113fc5450744f0f45d40859fddb99c2688))

## [1.2.0](https://github.com/okp4/faucet-web/compare/v1.1.1...v1.2.0) (2023-06-09)


### Features

* **faucet:** add captcha ([d856713](https://github.com/okp4/faucet-web/commit/d85671314ebed64f63be50b2ae670fabaf9f26c7))

## [1.1.1](https://github.com/okp4/faucet-web/compare/v1.1.0...v1.1.1) (2023-06-09)


### Bug Fixes

* **release:** remove commit-analyser release rules ([615417e](https://github.com/okp4/faucet-web/commit/615417ede1234d2aa06f7f9e9045acc0fa87eef8))
* **style:** replace id selectors ([1837e3b](https://github.com/okp4/faucet-web/commit/1837e3b91e524011ba24ed4d387292c34f24ffce))

## [1.1.0](https://github.com/okp4/faucet-web/compare/v1.0.6...v1.1.0) (2022-10-24)


### Features

* add version to html metadata ([c4c37e5](https://github.com/okp4/faucet-web/commit/c4c37e5edfd19b2da6c425707dad363cf79e8e3c))
* display current version in footer ([2088f3a](https://github.com/okp4/faucet-web/commit/2088f3a58ccbd61779bcc18dc3538729770c52d7))

## [1.0.6](https://github.com/okp4/faucet-web/compare/v1.0.5...v1.0.6) (2022-10-21)


### Bug Fixes

* change footer text color to a remaining white color ([d6038a5](https://github.com/okp4/faucet-web/commit/d6038a5f626f8470b4a9c3a03c81dbde272e3b95))
* **release:** rearrange conventionalcommits preset ([ab6d00a](https://github.com/okp4/faucet-web/commit/ab6d00a1022a52c5fa982ca724693463f93fb17f))

## [1.0.5](https://github.com/okp4/faucet-web/compare/v1.0.4...v1.0.5) (2022-09-07)

## [1.0.4](https://github.com/okp4/faucet-web/compare/v1.0.3...v1.0.4) (2022-08-31)

## [1.0.3](https://github.com/okp4/faucet-web/compare/v1.0.2...v1.0.3) (2022-08-17)

## [1.0.2](https://github.com/okp4/faucet-web/compare/v1.0.1...v1.0.2) (2022-08-05)

## [1.0.1](https://github.com/okp4/faucet-web/compare/v1.0.0...v1.0.1) (2022-07-20)

# 1.0.0 (2022-07-18)


### Features

* add Faucet component from @okp4/ui and add it to content ([2b95a0f](https://github.com/okp4/faucet-web/commit/2b95a0fc082dfa32f51fc02e0479953ff960445a))
* add footer component ([df0d65e](https://github.com/okp4/faucet-web/commit/df0d65e4f1ca9bc98414930bfd7e5c82b9fef76f))
* add okp4 logo for browser tab icon ([9edf4ad](https://github.com/okp4/faucet-web/commit/9edf4adb25115c10ef2f32f43f61d0c0decf3d63))
* add StoreProvider for SSR bypass and use it at root component ([733a389](https://github.com/okp4/faucet-web/commit/733a38926dbd4e135b5c3112628aa3227b6ca9bf))
* allow set config at runtime ([09c29d5](https://github.com/okp4/faucet-web/commit/09c29d5542cbc0d17e27e485813e4a1bea973392))
* configure client through config endpoint ([82e399c](https://github.com/okp4/faucet-web/commit/82e399cd4da505ff64fa30c78b70977fbc517cf3))
* declare keplrChainConfig object ([38724c8](https://github.com/okp4/faucet-web/commit/38724c8c4d2ec84a2a53621c68a4fb87beab0504))
* embeds header ([bdff29f](https://github.com/okp4/faucet-web/commit/bdff29fbe74ab0699039db00d3d7a7e158732459))
* expose config through rest endpoint ([98b7410](https://github.com/okp4/faucet-web/commit/98b74104aaef20522641da6586dd89208e30f68f))
* improve dynamic imports usage ([a478303](https://github.com/okp4/faucet-web/commit/a4783032d33971da5fb38975445b56665511d7e3))
* improve meta tag ([33b6295](https://github.com/okp4/faucet-web/commit/33b6295d3ef92e66d2c553ac5b0cbe822c3efce5))
* **redux:** init redux stores and export storeParameters ([8c20e55](https://github.com/okp4/faucet-web/commit/8c20e55f96ca94736faaa1023b94abba7a36a1ae))


src/config/chainConfig.ts
const keplrChainConfig = [
  {
    chainId: process.env.CHAIN_ID,
    chainName: process.env.CHAIN_NAME,
    rpc: process.env.CHAIN_RPC_ENDPOINT,
    rest: process.env.CHAIN_REST_ENDPOINT,
    bip44: {
      coinType: 118
    },
    bech32Config: {
      bech32PrefixAccAddr: 'okp4',
      bech32PrefixAccPub: 'okp4pub',
      bech32PrefixValAddr: 'okp4valoper',
      bech32PrefixValPub: 'okp4valoperpub',
      bech32PrefixConsAddr: 'okp4valcons',
      bech32PrefixConsPub: 'okp4valconspub'
    },
    currencies: [
      {
        coinDenom: 'KNOW',
        coinMinimalDenom: 'uknow',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'KNOW',
        coinMinimalDenom: 'uknow',
        coinDecimals: 6
      }
    ],
    stakeCurrency: {
      coinDenom: 'KNOW',
      coinMinimalDenom: 'uknow',
      coinDecimals: 6
    },
    coinType: 118,
    gasPriceStep: {
      low: 0.01,
      average: 0.025,
      high: 0.03
    }
  }
]

export default keplrChainConfig


src/config/keplr.ts
/* eslint-disable @typescript-eslint/prefer-readonly-parameter-types */
import type { Chain } from './config.type'

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
const keplrChainConfig = (chain: Chain) => [
  {
    chainId: chain.id,
    chainName: chain.name,
    rpc: chain.endpoints.rpc,
    rest: chain.endpoints.rest,
    bip44: {
      coinType: 118
    },
    bech32Config: {
      bech32PrefixAccAddr: 'okp4',
      bech32PrefixAccPub: 'okp4pub',
      bech32PrefixValAddr: 'okp4valoper',
      bech32PrefixValPub: 'okp4valoperpub',
      bech32PrefixConsAddr: 'okp4valcons',
      bech32PrefixConsPub: 'okp4valconspub'
    },
    currencies: [
      {
        coinDenom: 'KNOW',
        coinMinimalDenom: 'uknow',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'KNOW',
        coinMinimalDenom: 'uknow',
        coinDecimals: 6
      }
    ],
    stakeCurrency: {
      coinDenom: 'KNOW',
      coinMinimalDenom: 'uknow',
      coinDecimals: 6
    },
    coinType: 118,
    gasPriceStep: {
      low: 0.01,
      average: 0.025,
      high: 0.03
    }
  }
]

export default keplrChainConfig
